lib/edit-engine/addon/routes/application.js

```diff
@@ -0,0 +1,24 @@
+import Route from '@ember/routing/route';
+import { inject as service } from '@ember/service';
+
+export default class ApplicationRoute extends Route {
+  @service content;
+
+  model(params, transition) {
+    // return whatever model the transition has
+    return Object.entries(transition.resolvedModels).reduce((acc, item) => {
+      if (item[1]) { return item[1]; }
+    }, {});
+  }
+
+  renderTemplate(applicationController, model) {
+    const template = model.type;
+    const controller = this.controllerFor(model.type);
+    this.render(template, {
+      into: 'application', // i have not figured out how to get it to render into the engine's application template
+      controller,
+      model
+    });
```

Comment:
This is the key: As far as I can tell there is no way to mount a particular route of an engine at a particular route in the host. So we will always land at the edit-engine's application route.

If we then transition to a child route we get urls like documents/abc123/edit/document. Yuck.

But what we can do is leverage render/renderTemplate to decide what controller to render based on the model we got.